AlphaWallet/Core/Helpers/KeyboardChecker.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  KeyboardChecker.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 03.07.2020.
+//
+
+import UIKit
+
+extension UIApplication {
+    var bottomSafeAreaHeight: CGFloat {
+
+        let window = UIApplication.shared.windows[0]
+        let safeFrame = window.safeAreaLayoutGuide.layoutFrame
+
+        return window.frame.maxY - safeFrame.maxY
+    }
+}
+
+class KeyboardChecker: NSObject {
+
+    private weak var viewController: UIViewController?
+    private let notificationCenter = NotificationCenter.default
+    private let resetHeightDefaultValue: CGFloat
+
+    var constraint: NSLayoutConstraint?
+
+    init(viewController: UIViewController, resetHeightDefaultValue: CGFloat = -UIApplication.shared.bottomSafeAreaHeight) {
+        self.viewController = viewController
+        self.resetHeightDefaultValue = resetHeightDefaultValue
+        super.init()
+
+        notificationCenter.addObserver(self, selector: #selector(keyboardWillOpen), name: UIResponder.keyboardWillShowNotification, object: nil)
+        notificationCenter.addObserver(self, selector: #selector(keyboardWillHide), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+
+    private var additionalBottonInset: CGFloat {
```

Comment:
`Bottom` :)

Reply:
> So strange. But I thought I commented on this and posted a screenshot and you replied, but I can't find it anymore. Anyway, it is still happening if we toggle the software keyboard on/off (without hitting the Done button). But this could be due to a problem with the original layout? Because I can see it before this PR.
> 
> <img alt="" width="200" src="https://user-images.githubusercontent.com/56189/86504869-5f7c3e00-bdf0-11ea-898b-531f63c3d854.png">

@hboon i found how it happens. Maybe you can help me to resolve this issue? The reason is because in method `keyboardWillHide`, we set keyboard height to `0`. when actual keyboard height  is `114`,  the problem is that i cant detect right keybard height. when keyboard hides, it tells that it height equals `114` even it hides at all from screen.

Reply:
I don't know :)